lib/class-wp-rest-template-parts-controller.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * REST API: WP_REST_Template_Parts_Controller class
+ *
+ * @subpackage REST_API
+ * @package    WordPress
+ */
+
+/**
+ * Core class used to access menu templte parts via the REST API.
+ *
+ * @see   WP_REST_Controller
+ */
+class WP_REST_Template_Parts_Controller extends WP_REST_Posts_Controller {
+
+	/**
+	 * Retrieves a list of template parts.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		synchronize_theme_template_parts();
+
+		return parent::get_items( $request );
+	}
+
+	/**
+	 * Retrieves a single template parat.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		synchronize_theme_template_parts();
+
+		return parent::get_items( $request );
+	}
+}
```

Comment:
All this class controlled is needed just to call the "sync" function before the get requests. I was wondering if there were a simpler way to extend these API calls without requiring a custom class controller.

Reply:
I'd use the `rest_dispatch_request` filter and no-op if the incoming value is not null, otherwise do the synchronization and then return null from the filter.